packages/core/src/workers/password-encryption-worker.ts

```diff
@@ -0,0 +1,26 @@
+import path from 'node:path';
+
+import { Piscina } from 'piscina';
+
+/**
+ * User password encryption worker pool.
+ *
+ * @remarks
+ * This worker pool is used to encrypt user password with Argon2i encryption method.
+ * Since the encryption process is CPU intensive,
+ * to better protect the main thread I/O performance,
+ * we use separate thread threads to handle the encryption process.
+ */
+const passwordEncryptionWorker = new Piscina<string, string>({
+  // We cannot use `import.meta.url` here because the file structure differs between test and production builds.
+  // In production, the file is bundled, while in test mode, the original directory structure is preserved.
+  filename: path.join(process.cwd(), 'build/workers/tasks/argon2i.js'),
```

Comment:
use "fileURLToPath(import.meta.url)"?

just double check, the ESM sample in Piscina's README doesn't work?